torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -1247,6 +1248,31 @@ SegmentedGroup* SegmentCandidateFinder::mergeAllGivenGroups(
 
 namespace {
 
+inline void copyValue(
+    Val* key,
```

Comment:
Is the key guaranteed to be consistent across expression evaluators? If so can you remind me how?

Reply:
Thanks for pointing this out. Will rename the function and add more comment here. The way this function is used actually has the two expression evaluators pointing to the same fusion, and the produced `ExpressionEvaluator` will be used in `FusionSegmentGuard`'ed `Fusion` on a single `Segment`, which just re-defines the boundary of `Fusion` temporarily but the pointers are consistent. 

The segmenter now can actually work without this part as `FusionSegmentGuard` wouldn't stop the ExpressionEvaluator from using values that are on the same complete fusion but not within the segment. I will try to remove it in this PR.